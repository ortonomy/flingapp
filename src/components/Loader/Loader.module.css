.Loader {
  width: 50px;
  height: 50px;
  position: relative;
}

.LoaderElement {
  position: absolute;
  top: 0;
  left: 0;
}

.LoaderPip {
  width: 25px;
  height: 25px;
  left: 25px;
  transform-origin: bottom left;
  animation: loader 1s ease-in-out infinite;
}

@keyframes loader {
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
}