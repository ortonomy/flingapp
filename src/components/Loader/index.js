// standard react
import React from 'react';

//styles
import styles from './Loader.module.css';

// assets
import Oval from '../../assets/images/oval@svg.svg';
import Sector from '../../assets/images/sector@svg.svg';

const Loader = props => {
  return (
    <div className={styles.Loader}>
      <img className={[styles.LoaderBase, styles.LoaderElement].join(' ')} src={Oval} alt="" />
      <img className={[styles.LoaderPip, styles.LoaderElement].join(' ')} src={Sector} alt="" />
    </div>
  )
};

export default Loader;