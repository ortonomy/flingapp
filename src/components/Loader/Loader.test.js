import React from 'react';
import Loader from '../Loader';

it('renders without crashing', () => {
  const wrapper = shallow(<Loader />);
  expect(wrapper).toMatchSnapshot();
});
