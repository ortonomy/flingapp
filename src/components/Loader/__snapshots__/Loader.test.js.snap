// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt=""
          className=" "
          src="oval@svg.svg"
/>,
        <img
          alt=""
          className=" "
          src="sector@svg.svg"
/>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "className": " ",
          "src": "oval@svg.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "className": " ",
          "src": "sector@svg.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt=""
            className=" "
            src="oval@svg.svg"
/>,
          <img
            alt=""
            className=" "
            src="sector@svg.svg"
/>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "className": " ",
            "src": "oval@svg.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "className": " ",
            "src": "sector@svg.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
